### PR TITLE
Add jumpship recharge option to map right click menu

### PR DESCRIPTION
**Pull request by Skoraks**
Adds the option to fully charge the jump drive on the interstellar map in GM mode. (https://github.com/MegaMek/mekhq/issues/1078).
This is my first dip into Github and open source work so please let me know if I should be doing anything differently.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/CurrentLocation.java [hunk 1]
```diff
@@ -133,6 +133,13 @@ public void setJumpPath(JumpPath path) {
         jumpPath = path;
     }
 
+    public void setRecharged(Campaign campaign) {
+        DateTime currentDate = Utilities.getDateTimeDay(campaign.getCalendar());
+        double neededRechargeTime = currentPlanet.getRechargeTime(currentDate);
+        rechargeTime = neededRechargeTime;
+        campaign.addReport("Jumpship drives fully charged");
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/campaign/CurrentLocation.java]: Prefer moving this reporting logic out of this and into the action listener.

Also please add something like `(GM)` to the message.

**Comment by AaronGullickson** [comment]
This should wait until after PR #1280 because it will conflict with the refactoring of planets into planetary systems.

Also how are the rules applied to recharge the jump drive? Recharging is not instantaneous. These rules are covered on pg. 87 of *Strategic Operations*. Whenever possible we should follow the rules of the book.

File: MekHQ/src/mekhq/campaign/CurrentLocation.java (hunk 1 above)
Comment (Skoraks) [reply MekHQ/src/mekhq/campaign/CurrentLocation.java]: I think I've done what you wanted in e92de1337856ded47da3bf461191e11ddfd1c6be
Also I updated some of the existing report's charging typos

File: MekHQ/src/mekhq/campaign/CurrentLocation.java [hunk 2]
```diff
@@ -133,6 +133,13 @@ public void setJumpPath(JumpPath path) {
         jumpPath = path;
     }
 
+    public void setRecharged(Campaign campaign) {
+        DateTime currentDate = Utilities.getDateTimeDay(campaign.getCalendar());
```
[the remark this answers is not in the document]
Comment (Skoraks) [reply MekHQ/src/mekhq/campaign/CurrentLocation.java]: It does but not in the correct type, getDate() provides it as 'java.util.date', whereas getRechargeTime() is currently expecting a date time in 'org.joda.time.DateTime' - what do you recommend?

**Comment by Skoraks** [comment]
@AaronGullickson - Thanks for the heads up, as for the rules I have just added a GM mode only option to charge the drives fully at the moment, it's a cheat and not using any game mechanics.

**Comment by inqistor** [comment]
> This should wait until after PR #1280 because it will conflict with the refactoring of planets into planetary systems.
> 
> Also how are the rules applied to recharge the jump drive? Recharging is not instantaneous. These rules are covered on pg. 87 of _Strategic Operations_. Whenever possible we should follow the rules of the book.

It's for simulation of changing Jump Ships. Most Mercs don't have organic Jump Ships, so no reason to stick to the same one for whole travel. Technically, it should be easier to find free collar on Jump Ship jumping to one system along your path, that finding Jump Ship which will take you thorough next 6 jump points, and probably cross States Border at least once.
Anyway, IIRC charging from Recharging Stations is also not implemented. Everything is still based on Star spectrum.

**Comment by AaronGullickson** [comment]
Yes, I think that makes sense. The system refactoring is now in and I don't think this will actually conflict with that, but it does it looks like there are conflicts with some of the recent changes to the InterstellarMapPanel.

**Comment by sixlettervariables** [comment]
#1359 brings this up to speed.

**Pull request closed without merging**